### PR TITLE
Add io.github.ptitSeb.hydracastlelabyrinth build

### PR DESCRIPTION
**Pull request by fpiesche**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/ptitSeb/hydracastlelabyrinth/issues/28
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: io.github.ptitSeb.hydracastlelabyrinth.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    config-opts:
+      - -DUSE_SDL2=ON
+    post-install:
+      - install -Dm755 io.github.ptitSeb.hydracastlelabyrinth.sh /app/bin/io.github.ptitSeb.hydracastlelabyrinth.sh
+      - desktop-file-edit --set-key=Exec --set-value="flatpak run io.github.ptitSeb.hydracastlelabyrinth" /app/share/applications/io.github.ptitSeb.hydracastlelabyrinth.desktop
+      - install -Dm644 /run/build/hcl/io.github.ptitSeb.hydracastlelabyrinth.desktop /app/share/applications/io.github.ptitSeb.hydracastlelabyrinth-fullscreen.desktop
+      - desktop-file-edit --set-key=Name --set-value="Hydra Castle Labyrinth (Fullscreen)" /app/share/applications/io.github.ptitSeb.hydracastlelabyrinth-fullscreen.desktop
+      - desktop-file-edit --set-key=Exec --set-value="flatpak run io.github.ptitSeb.hydracastlelabyrinth -d" /app/share/applications/io.github.ptitSeb.hydracastlelabyrinth-fullscreen.desktop
+
+finish-args:
```
Comment (hfiguiere) [inline io.github.ptitSeb.hydracastlelabyrinth.yml:36]: this should be above modules.

File: io.github.ptitSeb.hydracastlelabyrinth.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  # hardware 3D
+  - --device=dri
+  # X11 + XShm access
+  - --share=ipc
+  - --socket=x11
+  - --socket=fallback-x11
```
Comment (hfiguiere) [inline io.github.ptitSeb.hydracastlelabyrinth.yml:42]: both are exclusive, if it supports wayland then `x11` should be removed.

File: io.github.ptitSeb.hydracastlelabyrinth.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    sources:
+      - type: git
+        url: https://github.com/ptitSeb/hydracastlelabyrinth.git
+        x-checker-data:
+          type: git
+          tag-pattern: ^v([\\d.]+)$
+      - type: script
+        dest-filename: io.github.ptitSeb.hydracastlelabyrinth.sh
+        commands:
+          - #!/bin/bash
```
Comment (hfiguiere) [inline io.github.ptitSeb.hydracastlelabyrinth.yml:23]:
```suggestion
```

File: io.github.ptitSeb.hydracastlelabyrinth.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+          - cd /app
+          - ./hcl $@
+    buildsystem: cmake
```
Comment (hfiguiere) [inline io.github.ptitSeb.hydracastlelabyrinth.yml:26]:
```suggestion
    buildsystem: cmake-ninja
```

File: io.github.ptitSeb.hydracastlelabyrinth.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    config-opts:
+      - -DUSE_SDL2=ON
+    post-install:
+      - install -Dm755 io.github.ptitSeb.hydracastlelabyrinth.sh /app/bin/io.github.ptitSeb.hydracastlelabyrinth.sh
+      - desktop-file-edit --set-key=Exec --set-value="flatpak run io.github.ptitSeb.hydracastlelabyrinth" /app/share/applications/io.github.ptitSeb.hydracastlelabyrinth.desktop
```
Comment (hfiguiere) [inline io.github.ptitSeb.hydracastlelabyrinth.yml:31]:
```suggestion
```

Definitely not. This should match what's in `command` more or less the arguments substitution

File: io.github.ptitSeb.hydracastlelabyrinth.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+      - install -Dm644 /run/build/hcl/io.github.ptitSeb.hydracastlelabyrinth.desktop /app/share/applications/io.github.ptitSeb.hydracastlelabyrinth-fullscreen.desktop
```
Comment (hfiguiere) [inline io.github.ptitSeb.hydracastlelabyrinth.yml:32]:
```suggestion
      - install -Dm644 io.github.ptitSeb.hydracastlelabyrinth.desktop /app/share/applications/io.github.ptitSeb.hydracastlelabyrinth-fullscreen.desktop
```

File: io.github.ptitSeb.hydracastlelabyrinth.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      - desktop-file-edit --set-key=Name --set-value="Hydra Castle Labyrinth (Fullscreen)" /app/share/applications/io.github.ptitSeb.hydracastlelabyrinth-fullscreen.desktop
+      - desktop-file-edit --set-key=Exec --set-value="flatpak run io.github.ptitSeb.hydracastlelabyrinth -d" /app/share/applications/io.github.ptitSeb.hydracastlelabyrinth-fullscreen.desktop
```
Comment (hfiguiere) [inline io.github.ptitSeb.hydracastlelabyrinth.yml:34]:
```suggestion
```

If you put `flatpak` here it's wrong.

**Comment by fpiesche** [comment]
Fixed the desktop file edits - I'm not sure why but for some reason I couldn't get my local build to generate the right shortcuts without those explicit `flatpak run` edits. Plausibly some kind of issue with my DE caching older, broken ones...?

File: io.github.ptitSeb.hydracastlelabyrinth.yml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+sdk: org.freedesktop.Sdk
+command: io.github.ptitSeb.hydracastlelabyrinth.sh
+
+finish-args:
+  # hardware 3D
+  - --device=dri
+  # X11 + XShm access
+  - --share=ipc
+  - --socket=x11
+  - --socket=fallback-x11
```
Comment (bbhtt) [inline io.github.ptitSeb.hydracastlelabyrinth.yml:13]: It's either x11 or (wayland and fallback-x11). Both does not make sense

Comment (hfiguiere) [reply io.github.ptitSeb.hydracastlelabyrinth.yml:13]: That's what I said 

see https://github.com/flathub/flathub/pull/4928#discussion_r1481517031

Comment (fpiesche) [reply io.github.ptitSeb.hydracastlelabyrinth.yml]: Ah, apologies - I'm not super familiar with how Wayland works so I completely misunderstood how these permissions work. Sorted now!

**Comment by hfiguiere** [comment]
bot, build io.github.ptitSeb.hydracastlelabyrinth

**Comment by fpiesche** [comment]
Build tested and working!

**Pull request closed without merging**